### DISCUSSION TITLE
What are your favorite plugins?

### DISCUSSION THREAD

**Discussion by hakusaro** [Pryaxis/TShock#1739] [General]
What are the plugins you can't live without?

**Comment by bartico6** [comment]
WorldEdit, hands down.

**Comment by Kojirremer** [comment]
**WorldEdit** is the most important, yes. Large scale world modification in-game is pretty useful.

As for other picks:
**AutoBroadcast** = Not only broadcasts timed/on demand messages, but it can also do the same with commands. Pretty useful.

**BanBuff** = Since TShock currently has no way to ban buffs as it does with items/tiles/projectiles, this supplements that. Useful for mount management.

**DieMob Regions** = A must when you want safe zones without disabling enemy spawns worldwide. Also, the "replace mobs" feature makes for a pretty sweet ecosystem editor.

**EssentialsPlus** = Various useful commands for management, like freezing/unfreezing time in-game, marking checkpoints, etc.

**InfiniteChests** = MarioE's initial version, outdated and all. Unfortunately the successors have some nasty issues that have kept me with the first one.

**InvSee** = The easiest way to check (and edit!) a player's SSC inventory.

**NameValidator** = A godsend to avoid users registering with chaotic characters.

**SEconomy** (also JIST and its child modules) = Since it's one of only a few currency plugins available _to the public_, the one most supported by other plugins, and its modules have a lot of other uses, right now I'm depending on these too much to ignore.

**WorldRegeneration** = Keeps the server fresh by rotating maps without having to be switching them manually.

And a couple more that, while not especially useful, makes the server either more fun or easier to manage:
_CommandRecipes / CustomItems / EmergencyRoom / LootTableEditor / PermaBuffs / RegionTrigger / SpawnMobX / HouseRegions_ (I wish there was an alternative that worked with TShock regions as well)

**Reply by hakusaro** [reply]
> MarioE's initial version, outdated and all. Unfortunately the successors have some nasty issues that have kept me with the first one.

What are the nasty issues?

**Reply by Kojirremer** [reply]
> > MarioE's initial version, outdated and all. Unfortunately the successors have some nasty issues that have kept me with the first one.
> 
> What are the nasty issues?


When trying to use dressers as chests, their tiles would expand an extra tile to the left, glitching it. Sometimes, users could modify the contents of refilling chests. Mimic spawning doesn't work, even though it was stated that it was supported.
Can't remember if there are more, but I'm pretty sure I reported them on their threads (v2 & v3).
